File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;
import entity.Result;

/**
 * @author Reolcharm
 *
 */
public interface BrandService {

	/**
	 * @return
	 * 查询所有品牌, 无分页
	 */
	List<TbBrand> findAll();
	
	/**
	 * 分页查询.封装结果集.
	 * @param pageNum 当前页码
	 * @param pageSize 每页显示条数
	 * @return
	 */
	PageResult findPage(Integer pageNum, Integer pageSize);
	
	/** 新增品牌功能
	 * @param tbBrand 
	 * @return 
	 */
	void add(TbBrand tbBrand);
	
	/** 编辑品牌时,进行数据回显功能
	 * @param id
	 * @return
	 */
	TbBrand findOne(Long id);
	
	/** 修改品牌功能
	 * @param tbBrand 
	 * @return 
	 */
	void update(TbBrand tbBrand);
	
	/**
	 * 删除品牌功能
	 * @param ids
	 */
	void delete(Long[] ids);
	
	/**
	 * 方法重载, 条件查询, 分页展示
	 * @param tbBrand
	 * @param page
	 * @param size
	 * @return
	 */
	PageResult findPage(TbBrand tbBrand, Integer page, Integer size);
	
	
	/**
	* 品牌下拉框数据
	*/
	List<Map> selectOptionList();
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
/* 自定义品牌服务层 */
app.service('brandService',
		function($http) {

			/* 显示所有品牌, 已过时 */
			this.findAll = function() {
				return $http.get("../brand/findPage.do");
			}

			/* page 當前頁 rows 每頁顯示條數; 分页显示所有品牌功能 */
			this.findPage = function(page, rows) {
				return $http.get('../brand/findPage.do?page=' + page + '&rows='
						+ rows);
			}

			/* 分页方法的重载. 混合提交 */
			this.search = function(page, rows, searchEntity) {
				return $http.post('../brand/findByExample.do?page=' + page
						+ '&rows=' + rows, searchEntity);
			}

			/* 数据回显功能 */
			this.findOne = function(id) {
				/* 发送 get请求,进行数据回显 */
				return $http.get('../brand/findOne.do?id=' + id);
			}

			/* 新增功能 */
			this.add = function(entity) {
				return $http.post('../brand/add.do', entity);
			}

			/* 修改后, 保存功能 */
			this.update = function(entity) {
				return $http.post('../brand/update.do', entity);
			}

			/* 删除功能 */
			this.dele = function(selectIds) {
				return $http.get("../brand/delete.do?ids=" + selectIds);
			}
		});